src/Razor/Microsoft.NET.Sdk.Razor/src/GenerateBlazorWebAssemblyBootJson.cs

```diff
@@ -0,0 +1,142 @@
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Runtime.Serialization.Json;
+using System.Text;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using ResourceHashesByNameDictionary = System.Collections.Generic.Dictionary<string, string>;
+
+namespace Microsoft.AspNetCore.Razor.Tasks
+{
+    public class GenerateBlazorWebAssemblyBootJson : Task
+    {
+        [Required]
+        public string AssemblyPath { get; set; }
+
+        [Required]
+        public ITaskItem[] Resources { get; set; }
+
+        [Required]
+        public bool DebugBuild { get; set; }
+
+        [Required]
+        public bool LinkerEnabled { get; set; }
+
+        [Required]
+        public bool CacheBootResources { get; set; }
+
+        public ITaskItem[] ConfigurationFiles { get; set; }
+
+        [Required]
+        public string OutputPath { get; set; }
+
+        public override bool Execute()
+        {
+            using var fileStream = File.Create(OutputPath);
+            var entryAssemblyName = AssemblyName.GetAssemblyName(AssemblyPath).Name;
+
+            try
+            {
+                WriteBootJson(fileStream, entryAssemblyName);
+            }
+            catch (Exception ex)
+            {
+                Log.LogErrorFromException(ex);
+            }
+
+            return !Log.HasLoggedErrors;
+        }
+
+        // Internal for tests
+        public void WriteBootJson(Stream output, string entryAssemblyName)
+        {
+            var result = new BootJsonData
+            {
+                entryAssembly = entryAssemblyName,
+                cacheBootResources = CacheBootResources,
+                debugBuild = DebugBuild,
+                linkerEnabled = LinkerEnabled,
+                resources = new ResourcesData(),
+                config = new List<string>(),
+            };
+
+            // Build a two-level dictionary of the form:
+            // - assembly:
+            //   - UriPath (e.g., "System.Text.Json.dll")
+            //     - ContentHash (e.g., "4548fa2e9cf52986")
+            // - runtime:
+            //   - UriPath (e.g., "dotnet.js")
+            //     - ContentHash (e.g., "3448f339acf512448")
+            if (Resources != null)
+            {
+                var resourceData = result.resources;
+                foreach (var resource in Resources)
+                {
+                    ResourceHashesByNameDictionary resourceList;
+
+                    var fileName = resource.GetMetadata("FileName");
+                    var extension = resource.GetMetadata("Extension");
+                    var resourceCulture = resource.GetMetadata("Culture");
+                    var assetType = resource.GetMetadata("AssetType");
+                    var resourceName = $"{fileName}{extension}";
+
+                    if (!string.IsNullOrEmpty(resourceCulture))
```

Comment:
We generate a different boot.json file for build and publish since the inputs are different. Calculating it in the task avoids duplicating this code in msbuild targets and unit testable.